app/src/main/java/org/mozilla/fenix/search/toolbar/SearchSelector.kt

```diff
@@ -21,9 +22,21 @@ internal class SearchSelector @JvmOverloads constructor(
 ) : RelativeLayout(context, attrs, defStyle) {
 
     private val binding = SearchSelectorBinding.inflate(LayoutInflater.from(context), this)
+    private var marginTop: Int = 0
```

Comment:
I am curious what kind of issues we ran into that required us to programmatically set the margin in the class. 
Could we also set the margin in the XML in `fragment_home` to avoid adding the code here?

Reply:
`SearchSelector` has a hardcoded top margin, that made it placing in alongside the [searchIcon](https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/res/layout/fragment_home.xml#138) difficult.

Reply:
This was a random lunch thought, but I haven't really looked into the code again with consideration of your comment yet. 

I am thinking in an ideal situation all of us would probably prefer to see all the layout properties defined in the XML layout. If another engineer was working on the search selector layout, they might be confused by where the top margin is coming from in their testing because they don't see the margin top 14dp defined in the XML. They would have to dig much deeper into the code in order to figure out that the margin is set manually in the code and that could be both surprising and maybe somewhat frustrating when working with this code. 

So, I am wondering if we could do some more exploring to see if we could define this margin in the XML. A naive idea I have would be to try `marginTop=-14dp` and see if that works? I don't necessarily have the answers right now, but I will take some time to play around with your PR to see if there is a reasonable solution when I have a chance. 

I am also happy to accept this as a followup investigation and it is possible that the only solution is to set the margin programatically. I just want to make sure we put some more thought into this since this will only improve how people work with this code.